src/renderer/store/me.store.ts

```diff
@@ -64,12 +95,197 @@ window.tachyon.onEvent("user/self", async (event) => {
+    }
+});
+
+window.tachyon.onEvent("friend/requestCancelled", async (event) => {
+    if (event?.from) {
+        me.incomingFriendRequestUserIds.delete(parseInt(event.from));
+        await unsubscribeFromUsers([event.from]);
+    }
+});
+
+window.tachyon.onEvent("friend/removed", async (event) => {
+    if (event?.from) {
+        me.friendUserIds.delete(parseInt(event.from));
+        await unsubscribeFromUsers([event.from]);
     }
 });
 
 // export const me = readonly(_me);
 export const auth = { login, playOffline, logout, changeAccount };
 
+// Friend methods
+export const friends = {
+    async sendRequest(to: string) {
+        const response = await window.tachyon.request("friend/sendRequest", { to });
+        if (response.status === "success") {
+            me.outgoingFriendRequestUserIds.add(parseInt(to));
+            await subscribeToUsers([to]);
+        }
+        return response;
+    },
+
+    async acceptRequest(from: string) {
+        const response = await window.tachyon.request("friend/acceptRequest", { from });
+        if (response.status === "success") {
+            me.incomingFriendRequestUserIds.delete(parseInt(from));
+            me.friendUserIds.add(parseInt(from));
+        } else {
+            // Remove from UI even on error since request is invalid
+            me.incomingFriendRequestUserIds.delete(parseInt(from));
+        }
+        return response;
+    },
+
+    async rejectRequest(from: string) {
+        const response = await window.tachyon.request("friend/rejectRequest", { from });
+        if (response.status === "success") {
+            me.incomingFriendRequestUserIds.delete(parseInt(from));
+            await unsubscribeFromUsers([from]);
+        } else {
+            // Remove from UI even on error since request is invalid
+            me.incomingFriendRequestUserIds.delete(parseInt(from));
+            await unsubscribeFromUsers([from]);
+        }
+        return response;
+    },
+
+    async cancelRequest(to: string) {
+        const response = await window.tachyon.request("friend/cancelRequest", { to });
+        if (response.status === "success") {
+            me.outgoingFriendRequestUserIds.delete(parseInt(to));
+            await unsubscribeFromUsers([to]);
+        } else {
+            // Remove from UI even on error since request is invalid
+            me.outgoingFriendRequestUserIds.delete(parseInt(to));
+            await unsubscribeFromUsers([to]);
+        }
+        return response;
+    },
+
+    async remove(userId: string) {
+        const response = await window.tachyon.request("friend/remove", { userId });
+        if (response.status === "success") {
+            me.friendUserIds.delete(parseInt(userId));
+            await unsubscribeFromUsers([userId]);
+        }
+        return response;
+    },
+
+    async fetchFriendList() {
+        const response = await window.tachyon.request("friend/list");
+        if (response.status === "success") {
```

Comment:
this if condition is unnecesary, there is a lot of unnecesary if conditions like this all over this file.